tests/Http/HttpRequestTest.php

```diff
@@ -122,7 +122,7 @@ public function testFullUrlMethod()
         $this->assertEquals('https://foo.com', $request->fullUrl());
 
         $request = Request::create('https://foo.com', 'GET');
-        $this->assertEquals('https://foo.com?coupon=foo', $request->fullUrlWithQuery(['coupon' => 'foo']));
+        $this->assertEquals('https://foo.com/?coupon=foo', $request->fullUrlWithQuery(['coupon' => 'foo']));
```

Comment:
Exactly, an explanation would be nice rather than just saying that there is a test for it.

In my opinion evidence points that this is a bug. How can these two things co-exist and not be a bug?

```
$request = Request::create('https://foo.com', 'GET');
$this->assertEquals('https://foo.com?a=b', $request->fullUrlWithQuery(['a' => 'b']));
```

and

```
$request = Request::create('https://foo.com?a=b', 'GET');
$this->assertEquals('https://foo.com/?a=b', $request->fullUrl());
```

It supposed to give you the same thing yet it doesn't.

I understand that this is a braking change for 5.2. Should we do it on 5.3 instead?